### PR TITLE
Support SDP validation xrefType

### PR DESCRIPTION
**Pull request by xichan-ms**
[AB#297519](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/297519)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6646)

### CODE REVIEW

File: src/docfx/template/TemplateEngine.cs [hunk 1]
```diff
@@ -3,9 +3,11 @@
 
 using System;
 using System.Collections.Concurrent;
+using System.Globalization;
 using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Xml.Schema;
```
Comment (xichan-ms) [inline src/docfx/template/TemplateEngine.cs:10]: will remove it in another commit.

File: src/docfx/template/TemplateEngine.cs [hunk 2]
```diff
@@ -204,6 +205,7 @@ private RenderType GetTocRenderType()
             }
 
             var jsonSchema = JsonUtility.DeserializeData<JsonSchema>(schemaString, new FilePath(schemaFilePath));
+            jsonSchema.SchemaType = mime.ToLower(new CultureInfo("en-US"));
```
Comment (xichan-ms) [inline src/docfx/template/TemplateEngine.cs:208]: >ToLower [](start = 41, length = 7)

will change to ToLowerInvariant

File: src/docfx/Errors.cs [hunk 3]
```diff
@@ -392,6 +392,12 @@ public static Error AtXrefNotFound(SourceInfo<string> source)
             public static Error XrefNotFound(SourceInfo<string> source)
                 => new Error(ErrorLevel.Warning, "xref-not-found", $"Cross reference not found: '{source}'.", source);
 
+            public static Error InvalidXrefType(SourceInfo<string> xref, string expectedXrefType, string realXrefType)
+               => new Error(ErrorLevel.Warning, "xref-invalid-type", $"Cross reference '{xref}' invalid type, which is expected '{expectedXrefType}' but now is '{realXrefType}'.", xref);
+
+            public static Error XrefTypeNotFound(SourceInfo<string> xref, string expectedXrefType)
+               => new Error(ErrorLevel.Warning, "xref-type-not-found", $"Cross reference '{xref}' type not found, which is expected '{expectedXrefType}'.", xref);
```
Comment (hejingkan2005) [inline src/docfx/Errors.cs:399]: >which is expected '{expectedXrefType}' [](start = 114, length = 38)

Change to - '{expectedXrefType}' is expected?

File: src/docfx/Errors.cs [hunk 4]
```diff
@@ -392,6 +392,12 @@ public static Error AtXrefNotFound(SourceInfo<string> source)
             public static Error XrefNotFound(SourceInfo<string> source)
                 => new Error(ErrorLevel.Warning, "xref-not-found", $"Cross reference not found: '{source}'.", source);
 
+            public static Error InvalidXrefType(SourceInfo<string> xref, string expectedXrefType, string realXrefType)
+               => new Error(ErrorLevel.Warning, "xref-invalid-type", $"Cross reference '{xref}' invalid type, which is expected '{expectedXrefType}' but now is '{realXrefType}'.", xref);
```
Comment (hejingkan2005) [inline src/docfx/Errors.cs:396]: >which is expected '{expectedXrefType}' but now is '{realXrefType}' [](start = 110, length = 66)

'{expectedXrefType}' is expected but now is '{realXrefType}'

Comment (xichan-ms) [reply src/docfx/Errors.cs:396]: OK

File: src/docfx/build/xref/XrefResolver.cs [hunk 5]
```diff
@@ -243,13 +244,54 @@ private void ValidateExternalXref()
...
+                if (!_internalXrefMap.Value.InternalXrefMapContainsKey(xrefGroup.Key))
                 {
                     _errorLog.Add(Errors.Xref.UidNotFound(xrefGroup.Key, xrefGroup.Select(xref => xref.ReferencedRepositoryUrl).Distinct()));
                 }
             }
         }
 
+        private void ValidateXrefType()
+        {
+            var a = _internalXrefMap.Value.GetXrefTypes();
```
Comment (hejingkan2005) [inline src/docfx/build/xref/XrefResolver.cs:256]: >var a = _internalXrefMap.Value.GetXrefTypes(); [](start = 12, length = 46)

What's for?

File: src/docfx/build/xref/InternalXrefMap.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            _internalXrefMap = internalXrefMap;
+            _xrefTypes = xrefTypes;
+        }
+
+        public IEnumerable<InternalXrefSpec[]> GetInternalXrefValues()
+        {
+            return _internalXrefMap.Values;
+        }
+
+        public bool InternalXrefMapContainsKey(string key)
```
Comment (hejingkan2005) [inline src/docfx/build/xref/InternalXrefMap.cs:26]: >InternalXrefMapContainsKey [](start = 20, length = 26)

Shorten to "ContainsKey()"?

**Pull request closed without merging**